coordinator/src/main/scala/filodb.coordinator/queryplanner/DefaultPlanner.scala

```diff
@@ -408,46 +408,67 @@ trait  DefaultPlanner {
     PlanResult(Seq(scalarBinaryExec))
   }
 
+  /**
+   * Function to optimize and convert `foo or vector(0)` like queries which would normally
+   * result in an expensive SetOperatorExec plan into a simple InstantFunctionMapper
+   * which is far more efficient.
+   */
+  def optimizeOrVectorZero(qContext: QueryContext, logicalPlan: BinaryJoin): Option[PlanResult] = {
```

Comment:
how about `optimizeOrVectorDouble`?

Reply:
+1 as  technically it can be ``foo or vector(10)``

Reply:
done